### DISCUSSION TITLE
Runtime error

### DISCUSSION THREAD

**Discussion by hablema** [zephir-lang/zephir#2282] [Q&A] [answered]
The stub.so file was created successfully. But when adding to php.ini , any PHP commands returns this error

Eg: php -v 

PHP Fatal error:  Class 'psr\http\message\messageinterface' not found in Unknown on line 0
Segmentation fault (core dumped)

PHP version 8.0

**Comment by Jeckerson** [comment]
Hello

With what version of Zephir it was compiled?

**Comment by hablema** [comment]
PHP 8.0.10 (cli) (built: Aug 26 2021 15:50:07) ( NTS )
Zephir Version 0.14.0
PHP Zephir Parser 1.3.8

PHPIZE Returns this :-

Configuring for:
PHP Api Version:         20200930
Zend Module Api No:      20200930
Zend Extension Api No:   420200930

**Comment by Jeckerson** [comment]
Make sure that you are using latest `beta.3` of Zephir `0.14.0`. Also make sure that `psr` extension is installed.

**Comment by hablema** [comment]
Zephir verision : Zephir 0.14.0 by Andres Gutierrez and Serghei Iakovlev (beta.3)

PSR was installed by : sudo apt-get install php8.0-psr

**Comment by Jeckerson** [comment]
What are steps that you execute?

**Comment by hablema** [comment]
1. Cloned Zephir from GIT
2. Composer Update
3. Cloned Zephir Parser from GIT
4. Generated the default Binary extension method
5. Added extsntion to PHP.ini
6. Zephir Init and added the default greeting class
7. Zephir Build
8. Added the generated stub.so to PHP.ini
9. Restarted apache
10 any php command after this give me this error

**Comment by Jeckerson** [comment]
Could you provide Zephir code example where you are using `psr\http\message\messageinterface` this interface?
And also generated `.zep.c` code of this specific file/class.

**Comment by hablema** [comment]
**greetings.zep**

```
namespace utils;
class Greeting
{
    public static function say()
    {
        echo "hello world!";
    }
}
```

**Comment by Jeckerson** [comment, marked as answer]
You mentioned `stub.so` at the beginning. According to your namespace, the generated file must be `utils.so`. It might be that you are building and compiling wrong project or folder.

**Reply by hablema** [reply]
@Jeckerson : Thank you. It worked flawlessly. The default documentation is a little confusing so I was building it in wrong. Thanks for helping me out.

**Comment by Jeckerson** [comment]
Try this tutorial and see if it works
https://docs.zephir-lang.com/0.12/en/tutorial#tutorial
